Error/ErrorHandler.php

```diff
@@ -26,13 +26,17 @@ class ErrorHandler
     /** @var string */
     private $internalErrorMessage;
 
-    public function __construct($internalErrorMessage = null, LoggerInterface $logger = null)
+    /** @var array */
+    private $exceptionMap;
+
+    public function __construct($internalErrorMessage = null, LoggerInterface $logger = null, array $exceptionMap = [])
     {
         $this->logger = (null === $logger) ? new NullLogger() : $logger;
         if (empty($internalErrorMessage)) {
             $internalErrorMessage = self::DEFAULT_ERROR_MESSAGE;
         }
         $this->internalErrorMessage = $internalErrorMessage;
+        $this->exceptionMap = $exceptionMap;
```

Comment:
Yes it could be done this way. But if these classes are set using a setter then `ErrorHandler` would have to reprocess `exceptionMap`.